server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/MailboxSetMethodContract.scala

```diff
@@ -5941,6 +5943,159 @@ trait MailboxSetMethodContract {
          |}""".stripMargin)
   }
 
+  @RepeatedTest(100)
+  def concurrencyChecksUponParentIdUpdate(server: GuiceJamesServer): Unit = {
```

Comment:
Not sure to get this test... You are trying to assign mailbox1 as parent of mailbox2 and mailbox2 as parent of mailbox1 concurrently right? Even with your loop detection logic I would not expect there is no update at all in both mailboxes (that's what you assert right?). Even with the comment at the end I don't understand well what is going on

Reply:
> You are trying to assign mailbox1 as parent of mailbox2 and mailbox2 as parent of mailbox1 concurrently right?

Yes

> Even with your loop detection logic I would not expect there is no update at all in both mailboxes (that's what you assert right?).

Both mailbox get's updated, data race happens ~50% of the time.

> Even with the comment at the end I don't understand well what is going on

What's going on is that the data is still accessible, nothing "blew up".